sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -137,13 +138,15 @@ def test_agglomerative_clustering():
                                              linkage=linkage)
         clustering.fit(X)
         # test caching
+        tempdir = mkdtemp()
         clustering = AgglomerativeClustering(
             n_clusters=10, connectivity=connectivity,
-            memory=mkdtemp(),
+            memory=tempdir,
             linkage=linkage)
         clustering.fit(X)
         labels = clustering.labels_
         assert_true(np.size(np.unique(labels)) == 10)
+        shutil.rmtree(tempdir)
```

Comment:
Maybe this would be better wrapped in a try / finally block? Under Python 3 we could use a `with TemporaryDirectory() as memory:` block but it does not exist in Python 2.